### ESMA 3016
### Edgar Acuna
### Lab22: Uso del modulo statsmodels para Inferencia acerca de la media.  Prueba z para  proporciones.
#### Octubre 2020

In [1]:
import numpy as np
import statsmodels.api as sm
import statsmodels.stats.api as sms
import statsmodels.formula.api as smf
from statsmodels.stats.proportion import proportions_ztest
import matplotlib.pyplot as plt
%matplotlib inline

Caso 1: Poblacion Normal y varianza ( 𝜎2 ) conocida

### Intervalo de confianza para la media 

Ejemplo: Intervalo de confianza para la media poblacional del colesterol de todos
 los pacientes con problemas cardiacos asumiendo que su distribucion es  normal
La varianza (o desviacion estandar) debe ser conocida. En este caso se asume que 
la desviacion estandar (sigma) es 13. Se usa la muestra de 20 pacientes

In [2]:
colest=[217,223,225,245,238,216,217,226,202,233,235,242,219,221,234,199,236,248,218,224] 
sm.stats.zconfint(colest,alpha=.10)

(221.0838759128287, 230.71612408717132)

In [3]:
#prueba de hipotesis
a,b=sm.stats.ztest(colest, value=220, alternative='larger')
print("El valor de la prueba estadistica es",a,"El p-values es:",b)

El valor de la prueba estadistica es 2.015030390280827 El p-values es: 0.021950732457882954


#### $H_o: \mu=220$
#### $H_a: \mu>220$

#### Caso 2: Poblacion Normal y varianza ($\sigma^2$) desconocida,muestra pequena
#### Intervalo de confianza para la media

La media muestral $\bar{x} es un estimador de la media poblacional $\mu$

Intervalo de confianza para $\mu$ del $100(1-\alpha)\%$, $\alpha$ es el nivel de significacion

$\bar{x}-t(\alpha/2,n-1)\frac{s}{\sqrt{n}},\bar{x}+t(\alpha/2,n-1)\frac{s}{\sqrt{n}}$

In [4]:
#Ejemplo 1: Hallar el intervalo de confianza del 95% para el promedio de todos los graduandos
#basado en la siguiente muestra de 14 graduandos
#Asumir que la poblacion es normal 
gpa=[3.15,2.80,2.56,3.17,3.73,2.77,3.18,3.12,3.33,2.99,2.71,2.83,3.45,3.55]

In [5]:
sms.DescrStatsW(gpa).tconfint_mean()

(2.898521565207428, 3.292907006221144)

In [6]:
#Ejemplo 7.5. Los tiempos de sobrevivencia (en años) de 12 personas que se han sometido a un transplante de
#corazón son los siguientes:
surv=[3.1,.9,2.8,4.3,.6,1.4,5.8,9.9,6.3,10.4,0,11.5]

In [7]:
#Hallar un intervalo de confianza del 99 por ciento para el promedio de vida
#de todas las personas que se han sometido a un transplante de corazón.
sms.DescrStatsW(surv).tconfint_mean(alpha=.01)

(1.122494220344997, 8.377505779655005)

#### Probar la hipotesis de que el tiempo de sobrevivencia promedio es mas de 4

In [8]:
sms.DescrStatsW(surv).ttest_mean(4,"larger")   

(0.642136781140391, 0.26696901006689233, 11.0)

#### El valor de la prueba estadistica es  0.64
#### El p-value es 0.2669
#### Los grados de libertad de la prueba t son 11

## Inferencia para proporciones

In [9]:
#Ejemplo 7.7. En 1990 en un cierto país, se reportó que dos de cada 5 personas pensaban que
#debería incrementarse el poder nuclear.  En una encuesta reciente hecha en
#1996 a  1225 personas se encontró que 478 de ellos pensaban que se debería
#aumentar el poder nuclear.  ¿Piensa Ud. que hay evidencia de que
#la opinión de la gente en 1996 ha cambiado con respecto a 1990? Justificar su contestación.
prueba,p_value=proportions_ztest(478,1225,.4)
print("Prueba=",prueba,"p-value=",p_value)

Prueba= -0.7028700551758291 p-value= 0.4821367341350553


#### La prueba estadistica de Z es -0.702 y el pvalue es 0.4821. No se rechaza la hipotesis Nula

In [10]:
#Ejemplo 7.8. El director de un hospital afirma que el 25 por ciento de los nacimientos que
#ocurren allí son por cesárea.  Un médico que trabaja en dicho hospital piensa
#que ese porcentaje es mayor.  Para probar su afirmación recolecta información
#de los 25 nacimientos ocurridos durante una semana. 
partos=['cesarea','normal','cesarea','normal','normal','normal','normal','cesarea','normal','cesarea',
        'normal','cesarea','normal','normal','normal','normal','normal','cesarea','normal','normal',
        'cesarea','normal','normal','cesarea','normal']

In [11]:
conteos=partos.count('cesarea')

In [12]:
trials=len(partos)

In [13]:
proportions_ztest(conteos,trials,value=.25,alternative='larger')

(0.7503063099984763, 0.226535121566539)

#### La prueba estadistica aproximada es 0.75 y el p-value es 0.226. No se rechaza la Hipotesis Nula